org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/NullTypeAnnotationTest.java

```diff
@@ -18797,6 +18797,36 @@ public void _testGH854() {
 		};
 	runner.runConformTest();
 }
+// duplicate of #1077
+public void testGH475() {
+	Runner runner = new Runner();
+	runner.testFiles =
+		new String[] {
+			"Controller.java",
+			"public class Controller<T> {\n" +
+			"    final static String ENDPOINT = \"controll\";\n" +
+			"}\n",
+			"RequestMapping.java",
+			"import java.lang.annotation.ElementType;\n" +
+			"import java.lang.annotation.Retention;\n" +
+			"import java.lang.annotation.RetentionPolicy;\n" +
+			"import java.lang.annotation.Target;\n" +
+			"\n" +
+			"@Target(ElementType.TYPE)\n" +
+			"@Retention(RetentionPolicy.RUNTIME)\n" +
+			"public @interface RequestMapping {\n" +
+			"	String name() default \"\";\n" +
+			"	String[] value() default {};\n" +
+			"}\n",
+			"CtlImpl.java",
+			"@RequestMapping(CtlImpl.CTL_ENDPOINT)\n" +
+			"public class CtlImpl extends Controller<String> {\n" +
+			"    final static String CTL_ENDPOINT = ENDPOINT + \"/ctl\";\n" +
+			"    static String value;\n" +
+			"}\n"
+		};
+	runner.runConformTest();
+}
```

Comment:
So, between this test and the set added for the part -1 fix for 4.28, have we made sure to include all snippets from the whole cluster of tickets ??

Reply:
Tests added / enabled:
* org.eclipse.jdt.core.tests.compiler.regression.NullTypeAnnotationTest.testGH854() for #854 
* org.eclipse.jdt.core.tests.compiler.regression.NullTypeAnnotationTest.testGH475() , oops, should say #476 
* org.eclipse.jdt.core.tests.model.CompletionTests3.testBug425035_method_c2() to challenge COQNR

Previously added via #1079:
* org.eclipse.jdt.core.tests.compiler.regression.NullTypeAnnotationTest.testGH1007_srikanth() for #1007
*  org.eclipse.jdt.core.tests.compiler.regression.NullTypeAnnotationTest.testGH969  oops, should say #986 
* org.eclipse.jdt.core.tests.compiler.regression.NullTypeAnnotationTest.testVSCodeIssue3076 for redhat-developer/vscode-java/issues/3076

Previously added via #630 
* org.eclipse.jdt.core.tests.compiler.regression.NullTypeAnnotationTest.testGH629_*

No test yet:
* #875 
* eclipse/eclipse.jdt.ls/issues/2386

Reply:
Sorry for messing up those numbers in test names. With 6 digit numbers there was only copy-paste. With 3 digits is tempting to just (mis)type them.